### ISSUE TITLE
Slotted holes missing

### ISSUE THREAD

**Issue by LAAU** [IfcOpenShell/IfcOpenShell#195]
Slotted holes missing when converts ifc to step or iges.
Seems that Ifcconvert support only round values for bolt diameter.
See attached IFC with slotted holes non-converted.

[Slotted holes missing.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/860747/Slotted.holes.missing.zip)

**Comment by aothms** [comment]
Thanks for the report. I can reproduce the problem. The issue manifests itself in all output formats, not just STEP or IGES. Unfortunately it will not be until next week that have time to investigate more.

**Comment by aothms** [comment]
The issue was due to the way the curved segments were joined to the straight segments. Where there somehow was a 0.025 mm gap. Which is notably bigger than the 1e-5 defined precision in the model. See below, where the circle has radius `10.3` and the points of the line segment are at `Y=-10.2749996185303`.

~~~
#53= IFCDIRECTION((0.,1.));
#54= IFCCARTESIANPOINT((15.,0.));
#55= IFCAXIS2PLACEMENT2D(#54,#53);
#56= IFCCIRCLE(#55,10.3);
#57= IFCTRIMMEDCURVE(#56,(IFCPARAMETERVALUE(0.)),(IFCPARAMETERVALUE(3.14159265358979)),.F.,.PARAMETER.);
#58= IFCCOMPOSITECURVESEGMENT(.CONTINUOUS.,.T.,#57);
#59= IFCCARTESIANPOINT((15.,-10.2749996185303));
#60= IFCCARTESIANPOINT((-15.,-10.2749996185303));
#61= IFCPOLYLINE((#59,#60));
#62= IFCCOMPOSITECURVESEGMENT(.CONTINUOUS.,.T.,#61);
~~~

Before the tolerance was hard limited to the modelling precision which caused the edges to loose the association to one of its vertices, which caused the trouble.

This has been fixed in f71ee35f577dd8658b1f71b48d84db9b92c2a626

If you run `IfcConvert` with the `-v` flag for verbosity you will see warnings when processing this file.

Note that due to fillet radius of the I-shape profile, the holes do not run entirely through. Some viewers do not model the fillet radius.

![lauu](https://cloud.githubusercontent.com/assets/1096535/24610394/a9638372-187e-11e7-94e1-40f7a6ed3d55.png)

**Comment by LAAU** [comment]
Thanks Thomas
Very good job, fast and accurate :)
